src/libsync/owncloudpropagator.cpp

```diff
@@ -1338,11 +1339,37 @@ void PropagateRootDirectory::slotSubJobsFinished(SyncFileItem::Status status)
         return;
     }
 
+    if (_hasError == SyncFileItem::NoStatus) {
+        switch (status) {
+        case SyncFileItem::NoStatus:
+        case SyncFileItem::FatalError:
+        case SyncFileItem::NormalError:
+        case SyncFileItem::SoftError:
+        case SyncFileItem::Conflict:
+        case SyncFileItem::FileIgnored:
+        case SyncFileItem::FileLocked:
+        case SyncFileItem::Restoration:
+        case SyncFileItem::FileNameInvalid:
+        case SyncFileItem::FileNameClash:
+        case SyncFileItem::DetailError:
+        case SyncFileItem::Success:
+            break;
+        case SyncFileItem::BlacklistedError:
+            _hasError = SyncFileItem::BlacklistedError;
+            break;
+        }
+    }
+
     propagator()->scheduleNextJob();
 }
 
 void PropagateRootDirectory::slotDirDeletionJobsFinished(SyncFileItem::Status status)
 {
+    if (_hasError != SyncFileItem::NoStatus && status == SyncFileItem::Success) {
+        qCInfo(lcPropagator) << "PropagateRootDirectory::slotDirDeletionJobsFinished" << "reporting previous error" << _hasError;
+        status = _hasError;
```

Comment:
@mgallien What is the concept behind this? I am not sure I fully understand. Could you elaborate?

Reply:
so if you look at the new automated test, you can see that we now check that deleted folders are deleted even if there are blacklisted errors during propagation
the idea is that we still want to do the folder deletions part of the propagation and if that part is succeeding, we still want to report a sync error at the end